### PR TITLE
Release 1.12: Version numbering and upgrade related changes

### PR DESCRIPTION
**Pull request by orhankislal**
Updates the version number to 1.12 for the release candidate.
Updates the changelists and other related files for upgrade.
Note that upgrade is not supported from versions prior to 1.9.1.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/153/

File: deploy/gppkg/madlib.spec.in [hunk 1]
```diff
@@ -7,7 +7,7 @@
 BuildRoot:      @MADLIB_GPPKG_RPM_SOURCE_DIR@
 Summary:        Apache MADlib for @GPDB_VARIANT@ Database
 License:        @CPACK_RPM_PACKAGE_LICENSE@
-Name:           Apache_MADlib
+Name:           madlib
```
Comment (edespino) [inline deploy/gppkg/madlib.spec.in]: This change doesn't appear to version or upgrade related. What is this change for (GPDB specific gppkg spec file)? Aren't we still delivering and rpm named Apache MADlib?

**Comment by edespino** [comment]
I believe there is a needed change in (doc/mainpage.dox.in) to add the previous version (v1.11.0) into the "User documentation for earlier releases" section.

File: deploy/gppkg/madlib.spec.in (hunk 1 above)
Comment (orhankislal) [inline deploy/gppkg/madlib.spec.in]: This change in name created some problems when trying to upgrade from 1.11 to 1.12. It was changed with the 1.11 release and it was not apparent until now. The rpm will be renamed to `apache-madlib-1.12-bin-Linux.rpm` as usual.

Comment (orhankislal) [inline deploy/gppkg/madlib.spec.in]: The change from `madlib` to `Apache_MADlib` created some problems during upgrade tests so I decided to revert it. The rpm will still be renamed to `apache-madlib-1.12-bin-Linux.rpm`

**Comment by edespino** [comment]
A couple of things to consider for future releases:
- As much as possible, document this process on the project's wiki (so anyone can pick up the work and not have to rely on so previous PRs or individual team member knowledge).
- Create a JIRA for this work so it will be apparent this work has to be completed before the release. Additionally, this will also allow it to be referenced in the commit message.

File: src/madpack/changelist.yaml [hunk 2]
```diff
@@ -9,27 +9,39 @@
...
 # in function name, return type, argument order or types, or removal of
 # the function. In each case, the original function is as good as removed and a
 # new function is created. In such cases, we should abort the upgrade if there
 # are user views dependent on this function, since the original function will
 # not be present in the upgraded version.
 udf:
-    # ----------------- Changes from 1.10.0 to 1.11 ----------
-    - __build_tree:
+    # ----------------- Changes from 1.11 to 1.12 ----------
+    - tree_train:
```
Comment (iyerr3) [inline src/madpack/changelist.yaml]: Are these necessary because of the change in parameter name?

Comment (orhankislal) [inline src/madpack/changelist.yaml]: Yes

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/185/

**Pull request closed without merging**